tests/runtime/service_generation/signature_parsing/test_docstrings.py

```diff
@@ -45,3 +85,67 @@ def test_extract_nested_type():
     assert _extract_nested_type("list(str)") == List[str]
     assert _extract_nested_type("List(SampleInputType)") == List[SampleInputType]
     assert _extract_nested_type("int") == None
+
+
+def test_is_optional_works_on_corner_cases_docstrings():
+    """
+    Test that is_optional works on different corner cases of docstrings for an optional field
+    """
+    # test docstring with multiple params of the same name still works, but
+    # evaluated whether "optional" on the first description
+    def _fn(self, some_input: Optional[int]):
+        """
+        Args:
+            some_input: yadda yadda blah int
+                optional int for input
+            some_input: I repeat myself
+
+        Returns:
+            None
+        """
+        pass
+
+    assert is_optional(_fn, "some_input") == True
+
+    # test if docstring_parser.parse throws an exception, we return None
+    with patch("docstring_parser.parse", side_effect=ParseError("mocked error")):
+        assert is_optional(_fn, "some_input") == None
```

Comment:
Nice, love to see it!

Do we also have test cases for places where the docstring parser fails but not with an exception? Like if there's no docstring defined at all or if it's an empty string?